packages/request/README.md

```diff
@@ -0,0 +1,362 @@
+
+Of course, any handler may choose to read and handle the stream, and return either no stream or a different stream in the process.
+
+</details>
+
+<details>
+  <summary><strong>Automatic Currying of Stream and Response</strong></summary><br>
+
+In order to simplify the common case for handlers which decorate a request, if `next` is called only a single time and `setResponse` was never called by the handler, the response set by the next handler in the chain will be applied to that handler's outcome. For instance, this makes the following pattern possible `return (await next(<req>)).data;`.
+
+Similarly, if `next` is called only a single time and neither `setStream` nor `getStream` was called, we automatically curry the stream from the future returned by `next` onto the future returned by the handler.
+
+Finally, if the return value of a handler is a `Future`, we curry `data` and `errors` as well, thus enabling the simplest form `return next(<req>)`.
+
+In the case of the `Future` being returned, `Stream` proxying is automatic and immediate and does not wait for the `Future` to resolve.
+
+</details>
+
+### Using as a Service
+
+Most applications will desire to have a single `RequestManager` instance, which can be achieved using module-state patterns for singletons, or for [Ember](https://emberjs.com) applications by exporting the manager as a [service](https://guides.emberjs.com/release/services/).
+
+*services/request.ts*
+```ts
+import RequestManager, { Fetch } from '@ember-data/request';
+import Auth from 'ember-simple-auth/ember-data-handler';
+
+export default class extends RequestManager {
+  constructor(createArgs) {
+    super(createArgs);
+    this.use([Auth, Fetch]);
+  }
+}
+```
+
+### Using with `@ember-data/store`
+
+To have a request service unique to a Store:
+
+```ts
+import Store from '@ember-data/store';
+import RequestManager, { Fetch } from '@ember-data/request';
+
+class extends Store {
+  requestManager = new RequestManager();
+
+  constructor(args) {
+    super(args);
+    this.requestManager.use([Fetch]);
+  }
+}
+```
+
+### Using with `ember-data`
+
+If using the package [ember-data](https://github.com/emberjs/data/tree/master/packages/-ember-data), the following configuration will automatically be done in order to preserve the legacy [Adapter](https://github.com/emberjs/data/tree/master/packages/adapter) and [Serializer](https://github.com/emberjs/data/tree/master/packages/serializer) behavior. Additional handlers or a service injection like the above would need to be done by the consuming application in order to make broader use of `RequestManager`.
+
+```ts
+import Store from '@ember-data/store';
+import RequestManager from '@ember-data/request';
+
+export default class extends Store {
+  requestManager = new RequestManager();
+
+  constructor(args) {
+    super(args);
+    this.requestManager.use([LegacyHandler]);
```

Comment:
I just can't see where this `LegacyHandler` is coming from

Reply:
Oh I see, the import didn't make it into the ReadMe. It's in the rfc iirc, will add here too.

Reply:
Sorry for not looking at the RFC, indeed I just see it there